mantidproject/mantid
# Project Recovery: No longer save the save algorithm in the recovery information

**Background of work**
Currently project recovery stores all history of workspaces in the files, this includes Saving to files, so the idea is to stop it from calling these saved files, the best way to leave the other algorithms still functioning normally was to stop it from saving specific files.

**Description of work.**
I added a list of algorithms that will be ignored by project recovery, this probably isn't in the best place but it saves transferring it across three algorithms which may be harder to maintain in the future.

**To test:**
- Load a workspace
- Load another workspace
- Group these workspaces
- Load another workspace (From a nexus file would be easier)
- Run SaveNexus on this workspace
- Wait for Project Recovery to save (using the debug level of logging in the logger)
- Run Segfault
- Reload mantid
- Click open script in scripting window it should only have the Load algorithm calls, and not the GroupWorkspaces or SaveNexus.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
